compiler/rustc_mir/src/transform/instrument_coverage.rs

```diff
@@ -1138,15 +1302,43 @@ fn filtered_terminator_span(terminator: &'a Terminator<'tcx>, body_span: Span) -
         // an `if condition { block }` has a span that includes the executed block, if true,
         // but for coverage, the code region executed, up to *and* through the SwitchInt,
         // actually stops before the if's block.)
-        TerminatorKind::Unreachable // Unreachable blocks are not connected to the CFG
+        TerminatorKind::Unreachable // Unreachable blocks are not connected to the MIR CFG
         | TerminatorKind::Assert { .. }
         | TerminatorKind::Drop { .. }
         | TerminatorKind::DropAndReplace { .. }
         | TerminatorKind::SwitchInt { .. }
-        | TerminatorKind::Goto { .. }
         // For `FalseEdge`, only the `real` branch is taken, so it is similar to a `Goto`.
+        // FIXME(richkadel): Note that `Goto` was moved to it's own match arm, for the reasons
+        // described below. Add tests to confirm whether or not similar cases also apply to
+        // `FalseEdge`.
         | TerminatorKind::FalseEdge { .. } => None,
 
+        // FIXME(richkadel): Note that `Goto` was initially filtered out (by returning `None`, as
```

Comment:
Might be worth opening an issue to track all the kinds of "span weirdness" encountered and collect FIXMEs like this. I'd like to see some discussion about what we could change in MIR building / other upstream code.

Reply:
Got it. I'll create an issue for this (and look for others to add) and add Issue references in the code once they are created.

Reply:
Issue #78542 (I'm adding this to the comment)